src/client/components/SocialButtons.tsx

```diff
@@ -47,75 +59,94 @@ const iconOverrides = css`
   }
 `;
 
-const Gap = () => (
+const Gap = ({ isNativeApp }: { isNativeApp: IsNativeApp }) => (
   <span
     css={css`
       width: 0;
       height: ${space[3]}px;
-      ${from.tablet} {
+      ${!isNativeApp &&
+      `${from.tablet} {
         width: ${space[3]}px;
         height: 0;
-      }
+      }`}
     `}
   ></span>
 );
 
-export const SocialButtons = ({
+const SocialButton = ({
+  label,
+  icon,
+  socialProvider,
   queryParams,
-  marginTop,
+  showGap,
+  isNativeApp,
 }: SocialButtonProps) => {
   return (
-    <div css={containerStyles(marginTop)}>
+    <>
       <LinkButton
         priority="tertiary"
         cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgGoogleBrand />}
+        icon={icon}
         href={buildUrlWithQueryParams(
           '/signin/:social',
           {
-            social: 'google',
+            social: socialProvider,
           },
           queryParams,
         )}
-        data-cy="google-sign-in-button"
-        data-link-name="google-social-button"
+        data-cy={`${socialProvider}-sign-in-button`}
+        data-link-name={`${socialProvider}-social-button`}
       >
-        Google
-      </LinkButton>
-      <Gap />
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={buttonOverrides}
-        icon={<SvgFacebookBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'facebook',
-          },
-          queryParams,
-        )}
-        data-cy="facebook-sign-in-button"
-        data-link-name="facebook-social-button"
-      >
-        Facebook
-      </LinkButton>
-      <Gap />
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgAppleBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'apple',
-          },
-          queryParams,
-        )}
-        data-cy="apple-sign-in-button"
-        data-link-name="apple-social-button"
-      >
-        Apple
+        {socialButtonLabel(label, isNativeApp)}
       </LinkButton>
+      {showGap && <Gap isNativeApp={isNativeApp} />}
+    </>
+  );
+};
+
+const socialButtonLabel = (label: string, isNativeApp?: IsNativeApp) => {
+  const capitalisedLabel = label.charAt(0).toUpperCase() + label.slice(1);
+  if (isNativeApp) {
+    return `Continue with ${capitalisedLabel}`;
+  } else {
+    return capitalisedLabel;
+  }
+};
+
+export const SocialButtons = ({
+  queryParams,
+  marginTop,
+  isNativeApp,
+}: SocialButtonsProps) => {
+  let buttonOrder;
+  if (isNativeApp === 'ios') {
+    buttonOrder = ['apple', 'facebook', 'google'];
+  } else if (isNativeApp === 'android') {
+    buttonOrder = ['google', 'facebook', 'apple'];
+  } else {
+    buttonOrder = ['google', 'facebook', 'apple'];
+  }
```

Comment:
This should probably be a method which returns the button order based on the `isNativeApp` parameter, rather than reassigning a variable. Makes it testable with a unit test too 😉 

```ts
const getButtonOrder = (isNativeApp?: IsNativeApp): string[] => {
  switch (isNativeApp) {
    case 'ios':
      return ['apple', 'facebook', 'google'];
    case 'android':
      return ['google', 'facebook', 'apple'];
    default:
      return ['google', 'facebook', 'apple'];
  }
}

// then it becomes
export const SocialButtons = ({
  queryParams,
  marginTop,
  isNativeApp,
}: SocialButtonsProps) => {
  const buttonOrder = getButtonOrder(isNativeApp);
...
```

Reply:
Much more functional and stops me from doing weird stuff to get around the immutable array checker, thanks a bunch 😁